frontend/server/src/Controllers/Contest.php

```diff
@@ -1059,10 +1061,25 @@ public static function getContestEditForTypeScript(
                 'problemsetNotFound'
             );
         }
-        $problems = \OmegaUp\DAO\ProblemsetProblems::getProblemsByProblemset(
+        [
+            'problems' => $problems,
+            'problemsAsObject' => $problemsAsObject,
+        ] = \OmegaUp\DAO\ProblemsetProblems::getProblemsByProblemset(
             $problemset->problemset_id
         );
 
+        // Adding the version log to every problem
+        $problemsWithVersion = [];
+        foreach ($problems as $index => $problem) {
+            $problemsWithVersion[] = array_merge(
+                \OmegaUp\Controllers\Problem::getVersions(
+                    $problemsAsObject[$index],
+                    $r->identity
+                ),
+                $problem
+            );
```

Comment:
```suggestion
        foreach ($problems as $problem) {
            $problemsWithVersion[] = array_merge(
                \OmegaUp\Controllers\Problem::getVersions(
                    new \OmegaUP\DAO\VO\Problems(
                        array_intersect_key(
                          $problem,
                          \OmegaUp\DAO\VO\Problems::FIELD_NAMES,
                        ),
                    ),
                    $r->identity
                ),
                $problem
            );
```

así no hay necesidad de tocar `\OmegaUp\DAO\ProblemsetProblems::getProblemsByProblemset` para nada(?) (aunque habrá que renombrar `log` por `versions`).

Reply:
Oh, esto va a reducir muchísimo el cambio. y no habrá necesidad de renombrar porque lo vamos a poner en otro nuevo elemento